### PR TITLE
Add uitest target tvos

### PR DESCRIPTION
**Pull request by sujata23**
Resolves [issue](https://github.com/tuist/tuist/discussions/3750)

### Short description 📝

> As discussed [here](https://github.com/tuist/tuist/discussions/3750) Adding support of uitest for tvOS Target

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.

### CODE REVIEW

**Comment by sujata23** [comment]
@fortmarek  I have added testcase for uiTest support for tvOS in this [file](projects/tuist/features/generate-8.feature) . But it is failing and showing following error. Can you give me some idea of how I can resolve this issue? 
Screenshot 2021-11-29 at 20 51 31

File: projects/tuist/fixtures/tvos_app_with_uitest/Project.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import ProjectDescription
+
+let project = Project(
+    name: "App",
+    targets: [
+        Target(
+            name: "App",
+            platform: .tvOS,
+            product: .framework,
```
Comment (ezraberch) [inline projects/tuist/fixtures/tvos_app_with_uitest/Project.swift:9]: This is why the test is failing.

```suggestion
            product: .app,
```

Comment (sujata23) [reply projects/tuist/fixtures/tvos_app_with_uitest/Project.swift]: ahh, it's my mistake. I have corrected it now. Thank you 😊

File: projects/tuist/features/generate-8.feature [hunk 2]
```diff
@@ -43,4 +43,11 @@ Scenario: The project is a macOS app with extensions (macos_app_with_extensions)
     Then I copy the fixture macos_app_with_extensions into the working directory
     Then I install the Workflow extensions SDK
     Then tuist generates the project
-    Then I should be able to build for macOS the scheme App
\ No newline at end of file
+    Then I should be able to build for macOS the scheme App
+
+Scenario: The project is an tvOS application with UItests (tvos_app_with_uitest)
```
Comment (danyf90) [inline projects/tuist/features/generate-8.feature:48]:
```suggestion
Scenario: The project is an tvOS application with UITests (tvos_app_with_uitest)
```

**Review by danyf90 — APPROVED** [review]

File: projects/tuist/features/generate-8.feature (hunk 2 above)
Comment (sujata23) [reply projects/tuist/features/generate-8.feature]: Done

**Review by fortmarek — APPROVED** [review]
LGTM - thanks a lot for this addition 👏

**Comment by fortmarek** [comment]
@all-contributors add @sujata23 for code

**Pull request merged**